### PR TITLE
Non evm solana support

### PR DESCRIPTION
**Pull request by shrinathprabhu**

### CODE REVIEW

File: src/components/ActivityView.vue [hunk 1]
```diff
@@ -90,19 +92,28 @@ function calculateCurrencyValue(valueInCrypto: bigint) {
...
+    const gasPrice = activity.transaction?.gasPrice || 0n
+    return activity.transaction.amount + gasUsed * gasPrice
   }
   return 0n
 }
 
 function getAmount(amount: bigint, isGas = false) {
-  if (isGas) {
-    return beautifyBalance(Number(ethers.utils.formatUnits(amount, 'gwei')), 5)
+  if (app.chainType === ChainType.solana_cv25519) {
```
Comment (mmjee) [inline src/components/ActivityView.vue:103]: Can we genericize this? Looks like we are hard-coding these conversions from the lowest denomination to the base denomination everywhere. Will be unscalable if we implement even more chain types.

File: src/components/UserWallet.vue [hunk 2]
```diff
@@ -45,16 +47,21 @@ type ModalState =
...
 const rpcStore = useRpcStore()
+const appStore = useAppStore()
 const showModal: Ref<ModalState> = ref(false)
 const { currency } = storeToRefs(rpcStore)
 const chainSelectedForEdit: Ref<number | null> = ref(null)
 const showAddressListDropDown = ref(false)
 
 const currencyStore = useCurrencyStore()
 const walletBalance = computed(() => {
+  let DecimalPow = 18
```
Comment (mmjee) [inline src/components/UserWallet.vue:58]: This should especially be genericized.

File: src/utils/solana/getPrivateKey.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import { etc, getPublicKey } from '@noble/ed25519'
+import { sha512 } from '@noble/hashes/sha512'
+import bs58 from 'bs58'
+
+// pk is in hex format with 64 length from key_helper
+function getPrivateKey(pk: string) {
```
Comment (mmjee) [inline src/utils/solana/getPrivateKey.ts:6]: You don't necessarily have to use the synchronous function, if it's feasible to make this asynchronous, you can just use `getPublicKeyAsync` and avoid this whole thing (it'll also use WebCrypto that way IIRC).

File: src/components/ActivityView.vue (hunk 1 above)
Comment (shrinathprabhu) [reply src/components/ActivityView.vue]: Fixed here [ba5fee8](https://github.com/arcana-network/wallet-ui/pull/535/commits/ba5fee89321cfaab746e450f6d3411ae17261c15)

File: src/components/UserWallet.vue (hunk 2 above)
Comment (shrinathprabhu) [reply src/components/UserWallet.vue]: Fixed here [ba5fee8](https://github.com/arcana-network/wallet-ui/pull/535/commits/ba5fee89321cfaab746e450f6d3411ae17261c15)

File: src/utils/solana/getPrivateKey.ts (hunk 3 above)
Comment (shrinathprabhu) [reply src/utils/solana/getPrivateKey.ts]: Fixed here https://github.com/arcana-network/wallet-ui/pull/535/commits/c1c34d09b26fc0a6774942137b4a20ff5fe9cbe4

File: src/components/ActivityView.vue (hunk 1 above)
Comment (shrinathprabhu) [reply src/components/ActivityView.vue]: Fixed here https://github.com/arcana-network/wallet-ui/pull/535/commits/ba5fee89321cfaab746e450f6d3411ae17261c15

File: src/components/UserWallet.vue (hunk 2 above)
Comment (shrinathprabhu) [reply src/components/UserWallet.vue]: Fixed here https://github.com/arcana-network/wallet-ui/pull/535/commits/ba5fee89321cfaab746e450f6d3411ae17261c15

**Review by makylfang — APPROVED** [review]

**Review by karthik-durai — APPROVED** [review]

**Pull request merged**